bokehjs/src/lib/core/util/spatial.ts

```diff
@@ -20,28 +20,20 @@ function upperBound(value: number, arr: ArrayLike<number>): number {
 
 class _FlatBush extends FlatBush {
 
-  get boxes(): TypedArray | Uint8ClampedArray {
-    return this._boxes
-  }
-
-  search_indices(minX: number, minY: number, maxX: number, maxY: number): Indices {
+  search_apply(minX: number, minY: number, maxX: number, maxY: number, nodeFunction: (index: number, node: Rect) => void): void {
     if (this._pos !== this._boxes.length) {
       throw new Error("Data not yet indexed - call index.finish().")
     }
 
     let nodeIndex: number | undefined = this._boxes.length - 4
     const queue = []
-    const results = new Indices(this.numItems)
 
     while (nodeIndex !== undefined) {
       // find the end index of the node
       const end = Math.min(nodeIndex + this.nodeSize * 4, upperBound(nodeIndex, this._levelBounds))
 
       // search through child nodes
       for (let pos: number = nodeIndex; pos < end; pos += 4) {
-        const index = this._indices[pos >> 2] | 0
-
-        // check if node bbox intersects with query bbox
```

Comment:
In the long term we shouldn't maintain this code altogether, especially given this is already fixed upstream. I would submit a PR to flatbush to add a second argument `node: Rect` to the current `search(..., filterFn(indices: number))`. We would use `filterFn` that returns `false` to avoid collecting results in an array. Then we would be able to remove all of this logic. flatbush is an active project for once, so let's take advantage of this.

Reply:
That sounds like a good idea but realistically I am not sure when I'd be able to do that myself.